easybuild/framework/easyblock.py

```diff
@@ -1678,6 +1679,59 @@ def checksum_step(self):
                 else:
                     raise EasyBuildError("Checksum verification for %s using %s failed.", fil['path'], fil['checksum'])
 
+    def check_checksums_for(self, ent, sub='', source_cnt=None):
```

Comment:
The `source_cnt` parameter seems buggy to me. It is used to limit the number of sources in extensions, right? But it actually doesn't do that if I understand the code correctly. You can have an extension with 2 sources, a single checksum, and it will be all fine, since we are passing `source_cnt = 1`, so the count is artificially limited to 1, but in reality 2 sources are listed. Then, there is just a single checksum, but we should have 2. Maybe it makes more sense to have a `is_extension = False` parameter, and if it is explicitly set to `True`, verify that the calculated `source_cnt` is exactly 1?

Reply:
There's a hard assumption in `easyblock.py` currently that each extension only has a single source file. The extension handling mechanism doesn't pick up on `sources` (so you can't specify a list of sources), it only provides `source_tmpl` to tweak the name of the source tarball for the extension (and it has a proper default for common type of extensions like Python packages, etc.).

That's why the `source_cnt=1` thing is being used here, since there simply isn't a list of `sources` to count for extensions...